iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/topology/RemoveHvdcLineTest.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.modification.topology;
+
+import com.powsybl.commons.test.AbstractConverterTest;
+import com.powsybl.iidm.modification.NetworkModification;
+import com.powsybl.iidm.network.DefaultNetworkListener;
+import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.test.HvdcTestNetwork;
+import com.powsybl.iidm.xml.NetworkXml;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.powsybl.iidm.modification.topology.TopologyTestUtils.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Anis Touri <anis-1.touri@rte-france.com>
+ */
+class RemoveHvdcLineTest extends AbstractConverterTest {
+
+    private final Set<String> removedObjects = new HashSet<>();
+    private final Set<String> beforeRemovalObjects = new HashSet<>();
+
+    @AfterEach
+    public void tearDown() {
+        removedObjects.clear();
+    }
+
+    private void addListener(Network network) {
+        network.addListener(new DefaultNetworkListener() {
```

Comment:
You don't need to test network listener. This is not the purpose of this network modification. Individual removal of hvdc line, converter stations, shunt compensator and listener impact is already tested somewhere else.